janusgraph-server/src/main/java/org/janusgraph/graphdb/server/JanusGraphServer.java

```diff
@@ -17,52 +17,91 @@
 import com.jcabi.manifests.Manifests;
 import org.apache.tinkerpop.gremlin.server.GremlinServer;
 import org.apache.tinkerpop.gremlin.server.Settings;
-import org.janusgraph.graphdb.server.utils.JanusGraphSettingsUtils;
+import org.janusgraph.graphdb.server.utils.GremlinSettingsUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.concurrent.CompletableFuture;
+
 public class JanusGraphServer {
     private static final Logger logger = LoggerFactory.getLogger(JanusGraphServer.class);
+    private GremlinServer gremlinServer = null;
+    private Settings gremlinSettings = null;
+    private final String confPath;
+    private CompletableFuture<Void> serverStarted = null;
+    private CompletableFuture<Void> serverStopped = null;
+
+    public JanusGraphServer(final String file) {
+        confPath = file;
+    }
 
     public static void main(final String[] args) throws Exception {
         printHeader();
-        final String file;
-        if (args.length > 0)
-            file = args[0];
-        else
-            file = "conf/janusgraph-server.yaml";
-
-        final Settings settings;
-        try {
-            settings = JanusGraphSettingsUtils.configureDefaultSerializersIfNotSet(Settings.read(file));
-        } catch (Exception ex) {
-            logger.error("Configuration file at {} could not be found or parsed properly. [{}]", file, ex.getMessage());
+        final String file = (args.length > 0) ? args[0] : "conf/janusgraph-server.yaml";
+        JanusGraphServer janusGraphServer = new JanusGraphServer(file);
+        if (!janusGraphServer.initialize()) {
```

Comment:
My pan is to reduce the code inside of main massively to increase testablity of full class.

Reply:
If you throw an exception in the initialize() method rather than return false, doesn't it make the main method even shorter?

That being said, if you intend to suppress exception stacktrace, then this is fair.

Reply:
I have to catch in the main or not. To get good logging behavior.